extras/docker/alpine3.18/Makefile

```diff
@@ -0,0 +1,23 @@
+MONGOCXX_VERSION=3.8.0
+
+.PHONY: build
+build:
+	docker build . \
+		-t mongo-cxx-driver:$(MONGOCXX_VERSION)-alpine3.18 \
+		--build-arg="MONGOCXX_VERSION=$(MONGOCXX_VERSION)"
+
+.PHONY: cleanbuild
+cleanbuild:
+	docker build . \
+		--no-cache \
+		-t mongo-cxx-driver:$(MONGOCXX_VERSION)-alpine3.18 \
+		--build-arg="MONGOCXX_VERSION=$(MONGOCXX_VERSION)"
```

Comment:
Since the image names are fixed, I would recommend splitting to a separate `clean` target, and use `docker rmi $(imagename) || :` to delete the image, and replace the `cleanbuild` with `clean build` plus a "weak dependency" from `build -> clean` (see other comment).

Reply:
In my opinion, doing a `--no-cache` build is a more idiomatic docker way to achieve this affect. It has the added benefit that if you are running this for local debugging, which is something that I intended with this approach, then it does not unexpectedly globally delete existing images on your machine.

Also, this suggestion has a bug. Suppose I'm currently running this image in a terminal, then at the same time I attempt to run a build for this image in the background, which is a workflow that I commonly find myself in. The call to delete the image will fail because the container using the image is currently running, for example see the error output below:
```
$ docker rmi mongo-cxx-driver:3.8.0-bookworm
Error response from daemon: conflict: unable to remove repository reference "mongo-cxx-driver:3.8.0-bookworm" (must force) - container 51d14d38441d is using its referenced image ce2044a5abb0
```
If I do a `--force rmi`, then it does not delete the image, but rather untags it, which means that if I run a build, then it will use the cached image again, which is not what is desired for a clean build.

If the name `cleanbuild` is confusing, then I can rename it to `nocachebuild`.

Reply:
This also would cause a race condition in CI if there are two builds happening on the same evergreen host. The two builds would share a docker daemon and encounter the same bug as described above if one of them is running the test at the same time that the other is building the image.